### PR TITLE
add network-instance for logging server

### PR DESCRIPTION
**Pull request by dplore**
### Change Scope

* Add network-instance to system logging to allow a remote syslog server to be accessed via a network-instance
* This change is backwards compatible
```
     +--rw logging
     |  +--rw console
     |  |  +--rw config
     |  |  +--ro state
     |  |  +--rw selectors
     |  |     +--rw selector* [facility severity]
     |  |        +--rw facility    -> ../config/facility
     |  |        +--rw severity    -> ../config/severity
     |  |        +--rw config
     |  |        |  +--rw facility?   identityref
     |  |        |  +--rw severity?   syslog-severity
     |  |        +--ro state
     |  |           +--ro facility?   identityref
     |  |           +--ro severity?   syslog-severity
     |  +--rw remote-servers
     |     +--rw remote-server* [host]
     |        +--rw host         -> ../config/host
     |        +--rw config
     |        |  +--rw host?               oc-inet:host
     |        |  +--rw source-address?     oc-inet:ip-address
     |        |  +--rw network-instance?   oc-ni:network-instance-ref
     |        |  +--rw remote-port?        oc-inet:port-number
     |        +--ro state
     |        |  +--ro host?               oc-inet:host
     |        |  +--ro source-address?     oc-inet:ip-address
     |        |  +--ro network-instance?   oc-ni:network-instance-ref
     |        |  +--ro remote-port?        oc-inet:port-number
     |        +--rw selectors
     |           +--rw selector* [facility severity]
     |              +--rw facility    -> ../config/facility
     |              +--rw severity    -> ../config/severity
     |              +--rw config
     |              |  +--rw facility?   identityref
     |              |  +--rw severity?   syslog-severity
     |              +--ro state
     |                 +--ro facility?   identityref
     |                 +--ro severity?   syslog-severity
```

### Platform Implementations

 * [Cisco IOS XR](https://www.cisco.com/c/en/us/td/docs/iosxr/cisco8000/system-monitoring/70x/b-system-monitoring-cg-cisco8k-70x/implementing_system_logging.html#concept_hgp_lj4_43b) supports logging to a remote server in a vrf
 * [Arista EOS](https://aristanetworks.force.com/AristaCommunity/s/article/logging-basic-syslog-and-beyond) supports logging to a remote server in a vrf
 * [JunOS](https://www.juniper.net/documentation/us/en/software/junos/junos-getting-started/topics/topic-map/management-interface-in-non-default-instance.html) supports logging in different / non-default network instances

### CODE REVIEW

File: release/models/system/openconfig-system-logging.yang [hunk 1]
```diff
@@ -426,6 +432,12 @@ module openconfig-system-logging {
         "Source IP address for packets to the log server";
     }
 
+    leaf network-instance {
```
Comment (rolandphung) [inline release/models/system/openconfig-system-logging.yang:435]: With the addition of `network-instance` it's possible that `host` as a key is not unique anymore.

**Comment by sulrich** [comment]
side note, there are deployments and implementations where the same IP address may have multiple ports in use.  would it be useful to change the remote-port to a leaf-list?

File: release/models/system/openconfig-system-logging.yang (hunk 1 above)
Comment (rolandphung) [inline release/models/system/openconfig-system-logging.yang:435]: Reminder that the lone addition of the `network-instance` will not allow duplicate IP addresses on different network instances even if it's a valid configuration. This is the same situation as being discussed in #767

File: release/models/system/openconfig-system-logging.yang [hunk 2]
```diff
@@ -426,6 +432,12 @@ module openconfig-system-logging {
...
+      type oc-ni:network-instance-ref;
+      description
+        "The network instance used to find the log server.";
```
Comment (jsterne) [inline release/models/system/openconfig-system-logging.yang:438]: Should we specify the behavior if the network-instance isn't configured? (i.e. what is the assumption today without this leaf? The default NI?).

**Review by robshakir — APPROVED** [review]
This LGTM.

**Comment by dplore** [comment]
This was reviewed in Jan 17, 2023 OC operators review meeting without objection.  Supporting multiple remote ports can be debated in  #790.

**Pull request merged**